### PR TITLE
ci: ✨ update CI pipeline

### PR DESCRIPTION
**Pull request by akakshuki**

### CODE REVIEW

File: .github/workflows/dotnet-ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  dotnet-build:
+    name: Run build dot Net project
```
Comment (zovippro1996) [inline .github/workflows/dotnet-ci.yml:11]:
```suggestion
    name: Run build dotnet project
```

File: .github/workflows/qac.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        run: |
+          dotnet restore
+          dotnet tool restore
+
+      - name: Run coverage code 
+        run: |
+          dotnet tool install --global coverlet.console
+          dotnet build --no-incremental
+          dotnet test --results-directory ./coverage-report --collect:"XPlat Code Coverage"
+          dotnet reportgenerator -reports:"./coverage-report/*/coverage.cobertura.xml" -targetdir:"./coverage-report" -reporttypes:"MarkdownSummary"
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:41]: Do we need this `dotnet reportgenerator` command in this pipeline ?

File: .github/workflows/qac.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      - name: Restore dependencies
+        run: |
+          dotnet restore
+          dotnet tool restore
+
+      - name: Run coverage code 
+        run: |
+          dotnet tool install --global coverlet.console
+          dotnet build --no-incremental
+          dotnet test --results-directory ./coverage-report --collect:"XPlat Code Coverage"
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:40]: Do we need this `dotnet test` if we already use coverlet cli ?

File: .github/workflows/qac.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            ${{ runner.os }}-dotnet-
+
+      - name: Restore dependencies
+        run: |
+          dotnet restore
+          dotnet tool restore
+
+      - name: Run coverage code 
+        run: |
+          dotnet tool install --global coverlet.console
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:38]: If this is required, should we add it to dotnet tool list? (.config), so developer can use the tool to generate opencover file?

File: .github/workflows/qac.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            -f=opencover \
+            -o="./coverage-report/coverage.xml"
+              
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report
+          path: ./coverage-report/*
+      
+      - name: Code coverage sumarise
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:55]:
```suggestion
      - name: Code coverage summarize
```

File: .github/workflows/qac.yml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+      - name: Download coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-report
+          path: ./coverage-report
+
+      - name: Sonarqube Analytis
+        run: |
+          # mv -f  ./coverage-report/*/coverage.cobertura.xml ./coverage-report/coverage.cobertura.xml
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:102]: Do we need this mv command? Should we delete it? ?

File: .github/workflows/qac.yml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar-${{ hashFiles('./.github/workflows/qac.yml') }}
+
+      - name: Download coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-report
+          path: ./coverage-report
+
+      - name: Sonarqube Analytis
```
Comment (zovippro1996) [inline .github/workflows/qac.yml:100]:
```suggestion
      - name: Sonarqube Analytics
```

File: .github/workflows/qac.yml (hunk 6 above)
Comment (akakshuki) [reply .github/workflows/qac.yml:102]: Will do

File: .github/workflows/qac.yml (hunk 4 above)
Comment (akakshuki) [reply .github/workflows/qac.yml:38]: This is specified for the code coverage report on Sonarqube. I don't think we should bring it to the developer.

File: .github/workflows/qac.yml (hunk 2 above)
Comment (akakshuki) [reply .github/workflows/qac.yml:41]: https://github.com/Planthor/PlanthorWebApi/actions/runs/8471923244

I would like to add the report to summarize  here

File: .github/workflows/qac.yml (hunk 3 above)
Comment (akakshuki) [reply .github/workflows/qac.yml:40]: Merged dotnet test with covelet cli

File: .github/workflows/qac.yml (hunk 4 above)
Comment (akakshuki) [reply .github/workflows/qac.yml:38]: Brought the tool to config file

File: .github/workflows/qac.yml (hunk 2 above)
Comment (zovippro1996) [reply .github/workflows/qac.yml]: Cool for me.

**Review by zovippro1996 — APPROVED** [review]

**Pull request merged**